src/ForeignKeyEditorDelegate.cpp

```diff
@@ -0,0 +1,162 @@
+            idsComboBox->setCurrentIndex(-1);
+            clauseEdit->clear();
+        });
+
+        connect(tablesComboBox, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+                [=](int)
+        {
+            idsComboBox->setCurrentIndex(-1);
+        });
+
+        connect(idsComboBox, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+                [=](int index)
+        {
+            bool enableClausesEditor = (index!= -1);
+            clauseEdit->setEnabled(enableClausesEditor);
+        });
+    }
+
+    QString getSql() const
+    {
+        if (tablesComboBox->currentText().isEmpty() || idsComboBox->currentText().isEmpty())
+            return QString();
+        else
+            return QString("%1(%2) %3")
+                .arg(sqlb::escapeIdentifier(tablesComboBox->currentText()))
+                .arg(sqlb::escapeIdentifier(idsComboBox->currentText()))
+                .arg(clauseEdit->text())
+                .trimmed();
+    }
+
+    QComboBox* tablesComboBox;
+    QComboBox* idsComboBox;
+    QLineEdit* clauseEdit; // for ON CASCADE and such
+
+private:
+    QPushButton* m_btnReset;
+};
+
+ForeignKeyEditorDelegate::ForeignKeyEditorDelegate(const DBBrowserDB& db, sqlb::Table& table, QObject* parent)
+    : QStyledItemDelegate(parent)
+    , m_db(db)
+    , m_table(table)
+{
+
+}
+
+QWidget* ForeignKeyEditorDelegate::createEditor(QWidget* parent, const QStyleOptionViewItem& option, const QModelIndex& index) const
+{
+    Q_UNUSED(option)
+    Q_UNUSED(index)
+
+    ForeignKeyEditor* editor = new ForeignKeyEditor(parent);
+    editor->setAutoFillBackground(true);
+
+    const auto objects = m_db.getBrowsableObjects();
+    for (auto obj : objects) {
+        if ("table" == obj.gettype()) {
+            QString tableName = obj.table.name();
+            if (tableName == m_table.name())
+                continue;
+
+            m_tablesIds.insert(tableName, obj.table.fieldNames());
+        }
+    }
+
+    editor->tablesComboBox->addItems(m_tablesIds.keys());
+
+    connect(editor->tablesComboBox, static_cast<void(QComboBox::*)(const QString &)>(&QComboBox::currentIndexChanged),
+            [=](const QString& tableName)
+    {
+        QComboBox* box = editor->idsComboBox;
+        box->clear();
+        box->addItems(m_tablesIds[tableName]);
+        box->setCurrentIndex(-1);
+    });
+
+    return editor;
+}
+
+void ForeignKeyEditorDelegate::setEditorData(QWidget* editor, const QModelIndex& index) const
+{
+    ForeignKeyEditor* fkEditor = static_cast<ForeignKeyEditor*>(editor);
+
+    int column = index.row(); // weird? I know right
+    sqlb::FieldPtr field = m_table.fields().at(column);
+    QSharedPointer<sqlb::ForeignKeyClause> fk = m_table.constraint({field}, sqlb::Constraint::ForeignKeyConstraintType).dynamicCast<sqlb::ForeignKeyClause>();
+    if (!fk.isNull()) {
+        fkEditor->tablesComboBox->setCurrentText(fk->table());
```

Comment:
Not sure about that. I just noticed that the column combo box doesn't get filled when the foreign key references the first table. A quick fix that I tried was to first call setCurrentIndex(-1) and then setCurrentText(...):
```c++
fkEditor->tablesComboBox->setCurrentIndex(-1);
fkEditor->tablesComboBox->setCurrentText(fk->table());
```
With this extra line it works just fine, but it's sort of a hack.